cedars/app/db.py

```diff
@@ -516,11 +516,41 @@ def get_all_annotations_for_patient(patient_id):
         "total": 0
     }
 
+    if unique_sentences:
+        # We first note the indices where duplicate sentences occur
+        prev_note_id = None
+        indices_to_remove = []
+        seen_sentences = set()
+        for i in range(len(annotations)):
+            # If we are on a new note, then the same sentence may be in a different context.
+            # So we only check for the same sentence in that note.
+            if annotations[i]['note_id'] != prev_note_id:
+                seen_sentences.clear()
+
+            prev_note_id = annotations[i]['note_id']
+            sentence = annotations[i]['sentence']
+            if sentence in seen_sentences:
+                indices_to_remove.append(i)
+                continue
+
+            seen_sentences.add(sentence)
+
+        # Remove the indices in reverse order to avoid a later index changing
+        # after a prior one is removed.
+
+        indices_to_remove.sort(reverse=True)
+        for index in indices_to_remove:
+            # Mark the annotation as reviewed before poping it
+            # This ensures that an unseen annotation cannot be unreviewed
+            mark_annotation_reviewed(annotations[index]["_id"])
+            annotations.pop(index)
+
+
     if len(annotations) > 0:
         result["annotations"] = [str(annotation["_id"]) for annotation in annotations]
         result["all_annotation_index"] = list(range(len(annotations)))
         # set array to 1 if annotation is unreviewed
-        result["unreviewed_annotations_index"] = [1 if not x["reviewed"] else 0 for x in annotations]
+        result["unreviewed_annotations_index"]= [1 if not x["reviewed"] else 0 for x in annotations]
```

Comment:
]= - seems to be a wrong formatting - please run the linter before submitting PRs